docs/design-docs/overview.md

```diff
@@ -0,0 +1,144 @@
+
+Here’re the details about running context and output structure for sb runner.
+
+**Context**
+
+```
+/path/to/working/directory
+├── outputs				# log directory
+│   └── 2020-12-18_20-43-42		# for each run, name in %Y-%m-%d_%H-%M-%S format
+│       ├── config.yaml		# dumped config object for internal use
+│       ├── ssh_config		# ~/.ssh/config for all hosts
+│       ├── id_rsa			# generated ssh private key for each run
+│       └── id_rsa.pub		# generated ssh public key for each run
+├── superbench				# local superbench code
+├── requirements.txt
+└── setup.py
+```
+
+**Output**
+
+```
+/path/to/working/directory
+└── outputs				# log directory
+    └── 2020-12-18_20-43-42		# for each run, name in %Y-%m-%d_%H-%M-%S format
+        ├── sb-run.log		# sb runner log
+        ├── node0			# directory for each node
+        │   ├── sb-exec.log		# sb executor log for each node
+        │   └── benchmarks		# results directory for benchmarks
+        │       ├── nccl		# nccl result
+        │       ├── pytorch_models	# pytorch models result
+        │       └── ...		# others
+        └── node...
+```
+
+### SB Executor
+
+SB executor is the component to run benchmarks inside Docker container. It will execute each benchmark actually and handle all pre- and post-processing, including health check, result validation, result processing, etc.
+
+Here’re the sb executor’s running context and output structure inside Docker container. The /root directory should be persistent on the host path.
+
+**Context**
+
+```
+/root
+├── .ssh				# ssh directory
+│   ├── config				# ssh_config from sb runner
+│   ├── id_rsa				# ssh private key from sb runner
+│   └── id_rsa.pub			# ssh public key from sb runner
+├── outputs				# log directory
+│   └── 2020-12-18_20-43-42		# for each run, same as sb runner
+│       └── config.yaml		# config object from sb runner
+├── superbench				# same superbench code as sb runner
+├── requirements.txt
+└── setup.py
+
+```
+
+**Output**
+
+```
+/root
+└── outputs				# log directory
+    └── 2020-12-18_20-43-42		# for each run, same as sb runner
+        ├── sb-exec.log		# sb executor log
+        └── benchmarks		# results directory for benchmarks
+            ├── nccl			# nccl result
+            ├── pytorch_models	# pytorch models result
+            └── ...			# others
+```
+
+### [Benchmarks](benchmarks.md)
+
+Benchmarks are a set of tests that actually run on compute node to measure the hardware performance. Here’re the related concepts, nccl benchmark is used as an example.
```

Comment:
compute node to node

Reply:
we only run tests on compute/gpu node?

Reply:
The key is we don't care about compute node or not. It's not the focus on this page.